m4/ax_cc_maxopt.m4

```diff
@@ -99,6 +99,23 @@ if test "$ac_test_CFLAGS" != "set"; then
          AX_CHECK_COMPILER_FLAGS(-mpower8-vector, CFLAGS="$CFLAGS -mpower8-vector")
          AX_CHECK_COMPILER_FLAGS(-mdirect-move, CFLAGS="$CFLAGS -mdirect-move")
      fi
+
+     if test -n "$TARGET_THUNDERX_COMPATIBLE"; then
+         CFLAGS="-Ofast"  # Override the above settings (they are not optimal for ThunderX)
+         AX_CHECK_COMPILER_FLAGS(-faggressive-loop-optimizations, CFLAGS="$CFLAGS -faggressive-loop-optimizations")
+         AX_CHECK_COMPILER_FLAGS(-fcrossjumping, CFLAGS="$CFLAGS -fcrossjumping")
+         AX_CHECK_COMPILER_FLAGS(-fdevirtualize, CFLAGS="$CFLAGS -fdevirtualize")
+         AX_CHECK_COMPILER_FLAGS(-fearly-inlining, CFLAGS="$CFLAGS -fearly-inlining")
+         AX_CHECK_COMPILER_FLAGS(-ffunction-cse, CFLAGS="$CFLAGS -ffunction-cse")
+         AX_CHECK_COMPILER_FLAGS(-fgcse-after-reload, CFLAGS="$CFLAGS -fgcse-after-reload")
+         AX_CHECK_COMPILER_FLAGS(-fforward-propagate, CFLAGS="$CFLAGS -fforward-propagate")
+         AX_CHECK_COMPILER_FLAGS(-fpack-struct, CFLAGS="$CFLAGS -fpack-struct")
+         AX_CHECK_COMPILER_FLAGS(-fsched-critical-path-heuristic, CFLAGS="$CFLAGS -fsched-critical-path-heuristic")
+         AX_CHECK_COMPILER_FLAGS(-fsel-sched-pipelining, CFLAGS="$CFLAGS -fsel-sched-pipelining")
+         AX_CHECK_COMPILER_FLAGS(-fselective-scheduling, CFLAGS="$CFLAGS -fselective-scheduling")
+         AX_CHECK_COMPILER_FLAGS(-ftrapping-math, CFLAGS="$CFLAGS -ftrapping-math")
```

Comment:
Are all of these flags necessary for the performance improvements, or is it only one or two of them that really matters?

Reply:
> Are all of these flags necessary for the performance improvements, or is it only one or two of them that really matters?

Among about a hundred of checked flags only those were left, that have statistically significant influence on performance for the target platform.

Reply:
Here are charts with several benchFFT tests for optimized vs default versions:
https://github.com/aur-ml/fftw3/wiki